### PR TITLE
fix(automation): hourly audit deltas, PR watchdog, and follow-up hardening

### LINKED ISSUE #1299: [review-followup] PR #1298: In coverage-comment job, fetching pull_request.head.sha by raw SHA can fail i...
Source PR: https://github.com/Keith-CY/carrier/pull/1298
PR title: fix: batch follow-up issue hardening (CI, credentials, smoke docs)

Auto-created from a non-blocking review suggestion marked with `NBS:`.

## Suggestion
- [ ] In coverage-comment job, fetching pull_request.head.sha by raw SHA can fail in some PR topologies (e.g. fork/shallow states). Consider guarding that fetch (or relying on base fetch + fallback path) so the workflow does not fail early under set -euo pipefail.

## Trace
- review:Akane-CN

### LINKED ISSUE #1300: [review-followup] PR #1298: `configv2/config.go` still contains its own `loadCredential`/`loadCredentialF...
Source PR: https://github.com/Keith-CY/carrier/pull/1298
PR title: fix: batch follow-up issue hardening (CI, credentials, smoke docs)

Auto-created from a non-blocking review suggestion marked with `NBS:`.

## Suggestion
- [ ] `configv2/config.go` still contains its own `loadCredential`/`loadCredentialFromKeychain`/`loadCredentialFromFile` implementations (simplified variants without error returns). Consider migrating those to use this shared package too for full deduplication.

## Trace
- review-comment:dev01lay2

### LINKED ISSUE #1301: docs: reconcile superseded plans with canonical architecture index
Automated hourly architecture audit detected superseded/diverged docs that require canonical linkage.

Repository: Keith-CY/carrier
Default branch: main

Findings:
- superseded/diverged docs: 4

Required actions:
1) Ensure each superseded/diverged doc has clear status + pointer to canonical sections.
2) Keep canonical architecture index aligned with implementation.
3) Remove ambiguity for contributors.

### LINKED ISSUE #1305: [review-followup] PR #1304: This change renames packaged release binary from agentd to carrier. If intent...
Source PR: https://github.com/Keith-CY/carrier/pull/1304
PR title: ci(release): ship binary-only artifacts named carrier

Auto-created from a non-blocking review suggestion marked with `NBS:`.

## Suggestion
- [ ] This change renames packaged release binary from agentd to carrier. If intentional, please update user-facing quickstart/install docs (and any downstream automation assumptions) in the same stream to avoid command mismatches after unzip.

## Trace
- review:Akane-CN

### LINKED ISSUE #1308: Auto-comment hourly audit deltas on open PRs
From hourly architecture audit suggestion backlog (2026-02-22).\n\nProposal:\n- When PR creation is skipped due to existing open PR, auto-comment latest audit delta into that PR (e.g. #1273).\n- Include key counters and trend changes so review context stays fresh.\n\nWhy:\n- Avoid silent skips\n- Keep reviewers synced without opening duplicate PRs/issues

### LINKED ISSUE #1309: Add docs_superseded trend tracking with regression-only alerts
Track docs_superseded as a rolling hourly metric (e.g. 24h).\n\nProposal:\n- Store historical values\n- Alert only when value increases (regression signal)\n- Keep normal/flat runs low-noise

### LINKED ISSUE #1310: Merge-readiness gate for consecutive clean runs
Proposal:\n- If docs_todo=0 and core_todo=0 for N consecutive runs (e.g. 3), mark PR as ready for final review.\n- Surface gate state in summary output.

### LINKED ISSUE #1311: Add test_files drift metric to hourly summary
Proposal:\n- Report test_files delta vs previous run.\n- Flag large unexpected drops/increases.\n\nGoal:\n- Catch silent test coverage movement early.

### LINKED ISSUE #1312: Stale-open-PR watchdog for blocked automation
Proposal:\n- Track age and unchanged diff/headSha for open PRs used by automation.\n- Escalate when PR remains unchanged beyond policy threshold.

### PR DESCRIPTION
**Pull request by Keith-CY**
## Summary
- extend hourly kanban sync with persisted audit metrics (`docs_todo`, `core_todo`, `docs_superseded`, `test_files`) and drift reporting
- add merge-readiness gate based on consecutive clean runs and expose gate state in hourly output
- add stale open PR watchdog using unchanged fingerprint tracking (`headSha + diff stats`) and stale thresholding
- upsert hourly delta comments on open automation audit PRs instead of silently skipping duplicate PR creation
- harden CI coverage baseline fetch to avoid raw head-SHA fetch failures on fork/shallow PR topologies
- deduplicate credential loader logic by reusing `daemon/credentialstore` in `configv2`
- add CI guard + tests to prevent reintroducing duplicate credential loader implementations
- align docs with `carrier` binary naming and strengthen canonical architecture linkage for historical plan docs

## Verification
- `node --test scripts/ci/*.test.cjs`
- `node scripts/classify-bucket.test.js && node scripts/kanban-project-id.test.js`
- `node scripts/ci/credential-loader-dedupe.cjs --root .`
- `bash scripts/ci/check-action-pinning.sh`
- `go test ./configv2`

Closes #1299
Closes #1300
Closes #1301
Closes #1305
Closes #1308
Closes #1309
Closes #1310
Closes #1311
Closes #1312

### CODE REVIEW

File: docs/deployment.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ For lifecycle state transitions, crash-loop behavior, and operator troubleshooti
 
 ```bash
 cd daemon
-go build -o agentd ./cmd/agentd
+go build -o carrier ./cmd/carrier
```
Comment (dev01lay2) [inline docs/deployment.md]: BS: `cd daemon` is kept but the build target was changed to `./cmd/carrier`. Under `daemon/` there is no `cmd/carrier` package (it lives at the repo root: `./cmd/carrier/main.go`). This should be:
```bash
go build -o carrier ./cmd/carrier
```
(remove `cd daemon`). Same issue was flagged on #1307.

File: scripts/ci/hourly-kanban-metrics.cjs [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+  computeDocsSupersededTrend,
+  computeMetricDrift,
+  computePrWatchdog,
+  encodeStateMarker,
+  extractStateFromBody,
+  isAutomationAuditPr,
+  normalizeState,
+  stripStateMarker,
+  upsertStateMarker,
+};
```
Comment (dev01lay2) [inline scripts/ci/hourly-kanban-metrics.cjs]: NBS: Consider adding `formatSignedDelta` to the module exports — it's used internally and could be useful for external consumers or testing.

**Review by Akane-CN — APPROVED** [review]
Reviewed and submitted approval.\n\nValidated changed automation paths and tests around:\n- hourly-kanban metrics/trend/watchdog logic\n- credential-loader dedupe validation\n- configv2 defaults cleanup\n\nReproduced targeted checks locally:\n- node --test scripts/ci/credential-loader-dedupe.test.cjs scripts/ci/hourly-kanban-metrics.test.cjs\n- go test ./configv2/... -count=1\n\nLGTM.

File: scripts/ci/hourly-kanban-metrics.cjs (hunk 2 above)
Comment (Keith-CY) [reply scripts/ci/hourly-kanban-metrics.cjs]: Kept `formatSignedDelta` internal for now; this helper is only used inside `buildAuditDeltaComment` in the same module. If we add an external consumer, I will export it together with a direct unit test for the exported contract.

File: docs/deployment.md (hunk 1 above)
Comment (Keith-CY) [reply docs/deployment.md]: Fixed in fc83fe9. Removed `cd daemon`, kept `go build -o carrier ./cmd/carrier` from repo root, and closed the bash code block before the Gateway section.

File: configv2/config.go [hunk 3]
```diff
@@ -164,10 +162,12 @@ func ApplyGatewayEnvironment(cfg *Config) error {
...
 			continue
 		}
-		if value, ok := loadCredential(m.CredentialRef); ok {
-			if err := setEnvIfUnset(m.EnvVar, value); err != nil {
-				return err
-			}
+		value, _, ok, err := credentialstore.LoadProviderCredential(m.CredentialRef)
+		if err != nil || !ok || strings.TrimSpace(value) == "" {
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (Keith-CY) [reply configv2/config.go]: Kept fail-open behavior intentionally for parity with previous `configv2` semantics: missing/unreadable credential store should not fail `ApplyGatewayEnvironment`; we only set env vars when a valid saved credential is available. This avoids startup failures caused by local credential-store state while preserving explicit env overrides.
